KenticoOnboardingApplication/src/KenticoOnboardingApplication.Api/Controllers/ListController.cs

```diff
@@ -17,38 +19,110 @@ public class ListController : ApiController
     {
         private readonly IListRepository _repository;
         private readonly IUrlLocator _urlLocator;
+        private readonly IUpdateItemService _updateItemService;
+        private readonly ICreateItemService _createItemService;
+        private readonly IGetItemService _getItemService;
 
-        public ListController(IListRepository repository, IUrlLocator locator)
+        public ListController(IListRepository repository, IUrlLocator locator, ICreateItemService createItemService,
+            IUpdateItemService updateItemService, IGetItemService getItemService)
         {
             _repository = repository;
             _urlLocator = locator;
+            _createItemService = createItemService;
+            _updateItemService = updateItemService;
+            _getItemService = getItemService;
         }
 
         public async Task<IHttpActionResult> GetAllItemsAsync() =>
             Ok(await _repository.GetAllItemsAsync());
 
         [Route("{id:guid}", Name = UrlLocator.RouteGet)]
-        public async Task<IHttpActionResult> GetItemAsync(Guid id) =>
-            Ok(await _repository.GetItemAsync(id));
+        public async Task<IHttpActionResult> GetItemAsync(Guid id)
+        {
+            ShouldBeIdEmpty(id, false);
+            if (!ModelState.IsValid)
+                return BadRequest(GetErrorMessage());
+            var result = await _getItemService.GetItemAsync(id);
+            if (!result.WasFound)
+                return NotFound();
+
+            return Ok(result.Item);
+        }
 
         public async Task<IHttpActionResult> PostItemAsync([FromBody] Item value)
         {
+            ValidateTextAndDateTimes(value);
+            ShouldBeIdEmpty(value.Id, true);
```

Comment:
If you need to use a flag (or any other "direct value", really), always use parameter name:
```suggestion
            ShouldBeIdEmpty(value.Id, shouldBeEmpty: true);
```

However, it is even better not use "flags" (or boolean switches) at all, see [ShouldBeIdEmpty](https://github.com/VeronikaLou/kentico-onboarding-cs/pull/4/files#diff-c2f34ff0b65e66e6945b2824ec988252R122).